### ISSUE TITLE
Merge grains errors

### ISSUE THREAD

**Issue by ajcross89** [mtex-toolbox/mtex#21]
Hi Ralf,

I’ve found a couple of small bugs in the ‘merge’ function. My code is as follows:

```
grains = calcGrains(ebsd,'angle',10*degree);

gb_qtz = grains.boundary('Quartz-new','Quartz-new');

rot = rotation('axis',Miller(0,0,0,1,CS{2}),'angle',60*degree);
ind = angle(gb_qtz.misorientation,rot)<2*degree; 
twinBoundary = gb_qtz(ind);

mergedGrains = merge(grains,twinBoundary);
```

The first error I get reads:

```
No public field meanRotation exists for class grain2d.

Error in grain2d/merge (line 47)
grains_merged.meanRotation = idRotation(numNewGrains,1);

Error in mergeGrainsDemo (line 69)
mergedGrains = merge(grains,twinBoundary);
```

Which I can disable by removing any lines containing `meanRotation` in the merge.m file (but then I can’t plot the IPF colours for merged grains).

The second error is when I remove some grains before merging:

```
grains = calcGrains(ebsd,'angle',10*degree);
grains = grains(grains.grainSize > 4);
```

And then the rest of the code as before. Removing small grains reduces the number of grains from 328 to 36. When running the merge function, I get this error:

```
Error using sparse
Index exceeds matrix dimensions.

Error in grain2d/merge (line 35)
A = sparse(mergeId(:,1),mergeId(:,2),1,length(grains),length(grains));

Error in mergeGrainsDemo (line 70)
mergedGrains = merge(grains,twinBoundary);
```

Any help would be great.

Thanks
Andrew

**Comment by gilgannonj** [comment]
Hello all,

I am having the same issue that Andrew had and I was wondering if it was ever resolved?

here's my code:

`% Override ebsd to exclude small grains
ebsd_restricted_median_3 = ebsd_restricted_median_3(selected_grains_restricted,'q')

% Get the grain boundaries
gb_qtz_restricted = selected_grains_restricted.boundary('Quartz','Quartz');

 % Define a rotation assosciated with dauphine twinning
rot_restricted = rotation('axis',Miller(0,0,0,1,ebsd_restricted_median_3('q').CS),'angle',60*degree);

% Calculate the angle between two rotations - the misorientation and the twinning rotation 
ang_restricted = angle(gb_qtz_restricted('Quartz','Quartz').misorientation,rot_restricted);

% Index all angles below a certain threshold
ind_ang_restricted = ang_restricted < 2.*degree;

% Select the boundaries based on this condition                                                                        
twinBoundary_restricted = gb_qtz_restricted(ind_ang_restricted);

% Merge grains connected with that twin boundary
[grains_merged_restricted,parentID_grains_restricted]= merge(selected_grains_restricted,twinBoundary_restricted);`

At this point I receive the same error as Andrew:

`Error using sparse
Index exceeds matrix dimensions.

Error in grain2d/merge (line 45)
A = sparse(mergeId(:,1),mergeId(:,2),1,length(grains),length(grains));`

I have used an EBSD data set that is indexed to have the condition that removes the small grains. I double checked I have the same number of grains in both the ebsd_restricted_median_3 and the selected_grains_restricted, and there are. In addition, twinBoundary_restricted is also not empty.

However if I just use the grains I extracted from the EBSD before any condition to exclude the small grains was applied, then I am allowed to merge the data. Even if I use the now overridden  ebsd_restricted_median_3 I am still allowed to merge. There only seems to be an issue if I try and use the grain data set that excludes small gains: selected_grains_restricted.

Did anyone ever find a way to get around this issue?

Best,

James
